src/components/card-list/CardList.tsx

```diff
@@ -1,95 +1,81 @@
 import type { ReactNode } from 'react';
-import { useState, useEffect } from 'react';
 import {
   useLocation,
   useSearchParams,
   Outlet,
-  Link,
   useNavigate,
 } from 'react-router-dom';
-import { NotFound } from '@/pages/not-found/NotFound.tsx';
+import { NotFound } from '@/pages/not-found/NotFound';
 import styles from './styles.module.scss';
-import type { IPeopleResponse } from '../../api/types.ts';
-import { useLocalStorage } from '../../hooks/useLocalStorage.ts';
-import { fetchData } from '../../api/api.ts';
-import { Card } from '../card/Card.tsx';
-import { Loader } from '../loader/Loader.tsx';
-import { Pagination } from '../pagination/Pagination.tsx';
+import { useLocalStorage } from '../../hooks/useLocalStorage';
+import { Card } from '../card/Card';
+import { Loader } from '../loader/Loader';
+import { Pagination } from '../pagination/Pagination';
+import { useFetchPeopleQuery } from '../../services/api.ts';
+import { Flyout } from '../flyout/Flyout.tsx';
 
 export function CardList(): ReactNode {
-  const [peoples, setPeoples] = useState<IPeopleResponse[]>([]);
-  const [isLoading, setIsLoading] = useState<boolean>(false);
-  const [maxPage, setMaxPage] = useState<number>(1);
-  const [statistic, setStatistic] = useState<string>('0 / 0');
-  const [searchParams, setSearchParams] = useSearchParams();
-  const navigate = useNavigate();
-
   const location = useLocation();
-  const cardsOnPage = 10;
-  const currentPage: number =
-    Number(new URLSearchParams(location.search).get('page')) || 1;
-
+  const [searchParams] = useSearchParams();
+  const navigate = useNavigate();
   const [searchQuery] = useLocalStorage('');
   const searchValue = searchParams.get('search') || searchQuery;
```

Comment:
Here is a problem.
Your app has an issue when you search for someone specific, then try to search with an empty string. It doesn't work.
The problem is that you use two instances of hook with state inside. Different instances keep different states. Hooks are not a Context to pass general data.
So here you could remove searchQuery and use data from search params everywhere you need it.
`const searchValue = searchParams.get('search') || '';`
```
        <Pagination
          numberPage={currentPage}
          maxPage={maxPage}
          searchValue={searchValue}
        />
```